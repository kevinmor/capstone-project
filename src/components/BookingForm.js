import { useState, useEffect } from 'react'

const BookingForm = ({availableTimes}) => {

    const handleSubmit = (e) => {
        e.preventDefault()
    }

    const formStyle = {
        display: "grid",
        maxWidth: "200px",
        gap: "20px"
    }

    const [date, setDate] = useState("")
    const [time, setTime] = useState("")
    const [guests, setGuests] = useState("")
    const [occasion, setOccasion] = useState("")

    const showTimes = availableTimes.map( i => <option key={i}>{i}</option>)

    useEffect(() => {
        console.log(availableTimes)
    }, [])


    return(
        <form onSubmit={handleSubmit} style={formStyle}>
            <label htmlFor="firstname">First name:</label>
            <input type="text" name="firstname" id="firstname" />

            
            <label htmlFor="res-date">Desired Reservation Date:</label>
            <input type="date" id="res-date" name="res-date" 
                value={date}
                onChange={(e) => setDate(e.target.value)}
            />

            
            <label htmlFor="res-time">Desired Time:</label>
            <select id="res-time"
                value={time}
                onChange={(e) => setTime(e.target.value)}
            >
            {showTimes}
            </select>

            <label htmlFor="guests">Number of guests:</label>
            <input type="number" placeholder="1" min="1" max="10" id="guests" 
                value={guests}
                onChange={(e) => setGuests(e.target.value)}
            />
            
            <label htmlFor="occasion">Occasion:</label>
            <select id="occasion"
                value={occasion}
                onChange={(e) => setOccasion(e.target.value)}
            >
                <option>Birthday</option>
                <option>Anniversary</option>
            </select>
            
            <input type="submit" value="Make your reservation" />
        </form>
    )
}

export default BookingForm