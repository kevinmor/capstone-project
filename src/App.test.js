import { render, screen } from "@testing-library/react";
import BookingForm from "./components/BookingForm";


test('Renders the BookingForm heading', () => {
    //need to add a prop to Booking form so that jest can simulate time slots being rendered, otherwise an error gets thrown
    const times = ['17:00', '18:00', '19:00', '20:00', '21:00', '22:00']
    render(<BookingForm availableTimes={times}/>);

    const headingElement = screen.getByText("Book Now");
    expect(headingElement).toBeInTheDocument();
})

test('Initial initiateTimes function should return array with times', () => {
    const initalizedExpected = ['17:00', '18:00', '19:00', '20:00', '21:00', '22:00']
    const initializeTimes = () => ['17:00', '18:00', '19:00', '20:00', '21:00', '22:00']
    expect(initializeTimes()).toMatchSnapshot(initalizedExpected)
})

test('Inital Update time function should return the same value provided to state', () => {
    const initalizedExpected = ['17:00', '18:00', '19:00', '20:00', '21:00', '22:00']
    const updateTimes = (state, action) =>{
        return state
    }
    expect(updateTimes(initalizedExpected)).toMatchSnapshot(initalizedExpected)
})

test('Function should return 3', () => {
    const add = (a,b) => a + b
    const a = 1
    const b = 2
})