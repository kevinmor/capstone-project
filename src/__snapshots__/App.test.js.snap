// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inital Update time function should return the same value provided to state 1`] = `
Array [
  "17:00",
  "18:00",
  "19:00",
  "20:00",
  "21:00",
  "22:00",
]
`;

exports[`Initial initiateTimes function should return array with times 1`] = `
Array [
  "17:00",
  "18:00",
  "19:00",
  "20:00",
  "21:00",
  "22:00",
]
`;
